internal/services/sources_service.go

```diff
@@ -0,0 +1,132 @@
+		return
+	}
+}
+
+func GetSource(w http.ResponseWriter, r *http.Request) {
+	id, err := ParseUint64(r, "ID")
+	if err != nil {
+		renderError(w, r, payloads.NewURLParsingError(r.Context(), "ID", err))
+		return
+	}
+
+	sourceDao, err := dao.GetSourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "source DAO", err))
+		return
+	}
+
+	source, err := sourceDao.GetById(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "get source by id", err))
+		return
+	}
+
+	if err := render.Render(w, r, payloads.NewSourceResponse(source)); err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "render source", err))
+	}
+}
+
+func CreateSource(w http.ResponseWriter, r *http.Request) {
+	payload := &payloads.SourceRequest{}
+	if err := render.Bind(r, payload); err != nil {
+		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), err))
+		return
+	}
+
+	// An ARN linked to the policy and role that are needed for our application
+	arn := os.Getenv("ARN")
+
+	sourceDao, err := dao.GetSourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "source DAO", err))
+		return
+	}
+
+	_, err = sources.CreateSource(w, r, payload.Name, arn, payload.OrgID)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "create source", err))
+		return
+	}
+
+	//create a Source object
+	source := &models.Source{
+		AuthID:   payload.AuthID,
+		OrgID:    payload.OrgID,
+		Name:     payload.Name,
+		SourceID: payload.SourceID,
+	}
+
+	// Add the source to the DB
+	source, err = sourceDao.Create(r.Context(), source)
+	if err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "create source in db", err))
+		return
+	}
+
+	if err := render.Render(w, r, payloads.NewSourceResponse(source)); err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "source", err))
+	}
+}
+
+func DeleteSource(w http.ResponseWriter, r *http.Request) {
+	payload := &payloads.SourceRequest{}
+	if err := render.Bind(r, payload); err != nil {
+		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), err))
+		return
+	}
+
+	sourceDao, err := dao.GetSourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "source DAO", err))
+		return
+	}
+
+	id, err := ParseUint64(r, "id")
+	if err != nil {
+		renderError(w, r, payloads.NewURLParsingError(r.Context(), "ID", err))
+		return
+	}
+
+	// Delete source from db
+	source, err := sourceDao.Delete(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewRenderError(r.Context(), "delete source from db", err))
+		return
+	}
+
+	// Delete source from sources
+	_, err = sources.DeleteSource(w, r, source.SourceID, payload.OrgID)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete source from sources", err))
+		return
+	}
+
```

Comment:
I would probably recommend first sources then db, so in case the request fails user can try again. Also empty line.

Reply:
I agree but I have a problem, I need to get the source_Id first in order to delete it in sources. 
I didn't want to query the DB twice - one to get the source_id and the other time to delete the source itself with the provided id. 
Do you have a better suggestion?